scipy/stats/_continuous_distns.py

```diff
@@ -10286,6 +10286,140 @@ def _single_cdf(q, k, df):
+    seeks to model the :math:`Z^0` boson with :math:`M \approx 91.1876 text{
+    GeV}` and :math:`\Gamma \approx 2.4952\text{ GeV}` [2]_ one can set `rho`
+    equal to ``91.1876/2.4952`` and ``scale=2.4952``.
+
+    To ensure a physically meaningful result when using the `fit` method, one
+    should set ``floc=0`` to fix the location parameter to 0.
+
+    References
+    ----------
+    .. [1] Relativistic Breit-Wigner distriution, Wikipedia,
+           https://en.wikipedia.org/wiki/Relativistic_Breit-Wigner_distribution
+    .. [2] M. Tanabashi et al. (Particle Data Group) Phys. Rev. D 98, 030001 -
+           Published 17 August 2018
+
+    %(example)s
+
+    """
+    def _argcheck(self, rho):
+        return rho > 0
+
+    def _shape_info(self):
+        return [_ShapeInfo("rho", False, (0, np.inf), (False, False))]
+
+    def _pdf(self, x, rho):
+        k = np.sqrt(
+            2 * (1 + 1/rho**2) / (1 + np.sqrt(1 + 1/rho**2))
+        ) * 2 / np.pi
+        return k / (((x**2 - rho**2)/rho)**2 + 1)
+
+    def _cdf(self, x, rho):
+        # Factor of 1/(2*rho) has been absorbed into the constant.
+        k = np.sqrt(
+                2 * (1 + 1/rho**2) / (1 + np.sqrt(1 + 1/rho**2))
+            ) / np.pi
+        result = np.arctan(x/np.sqrt(-rho*(rho + 1j))) / np.sqrt(-1 - 1j/rho)
+        # For real entries of x, one can take advantage of the quantity to be
+        # subtracted being the complex conjugate of the first. This is not the
+        # case for complex entries of x with nonzero imaginary part.
+        result -= _lazywhere(
+            np.iscomplex(x),
+            (x, result),
+            lambda x_, result_: np.arctan(x/np.sqrt(-rho*(rho - 1j))) /
+            np.sqrt(-1 + 1j/rho),
+            f2=lambda x_, result_: result.conjugate(),
+        )
+        result = - k * 1j * result
+        return result if np.iscomplexobj(x) else np.real(result)
+
+    def _munp(self, n, rho):
+        if n == 1:
+            # Factor of 1/(2*rho) has been absorbed into the constant.
+            k = np.sqrt(
+                2 * (1 + 1/rho**2) / (1 + np.sqrt(1 + 1/rho**2))
+            ) / np.pi * rho
+            return k * (np.pi/2 + np.arctan(rho))
+        if n == 2:
+            # Factor of pi/4 has been absorbed into the constant.
+            k = np.sqrt(
+                (1 + 1/rho**2) / (2 * (1 + np.sqrt(1 + 1/rho**2)))
+            ) * rho
+            result = (1 - rho * 1j) / np.sqrt(-1 - 1j/rho)
+            result += result.conjugate()
+            result = k * result
+            return np.real(result)
+        else:
+            return np.inf
+
+    def _stats(self, rho):
+        # Returning None from stats makes public stats use _munp.
+        return None, None, np.nan, np.nan
+
+    @inherit_docstring_from(rv_continuous)
+    def fit(self, data, *args, **kwds):
+        # Override rv_continuous.fit to better handle case where floc is set.
```

Comment:
Can you override `_fitstart` instead?

Reply:
Possibly. It depends on if we only want to support accurate fitting for the floc=0 case. Which very well may be true since this is the case that is relevant for applications. What ends up happening is that the initial point ends up biased to the floc=0 case, and because the likelihood landscape is highly multimodal, you’re liable to end up with a result close to the best floc=0 solution regardless of the data. As far as I can tell, _fitstart isn’t currently aware of the value of floc, and I’d be anxious about modifying such a crucial piece of distribution infrastructure. By overriding fit, we can make fitting work well with any value of floc. If floc isn’t set, fitting won’t work well in any case though.